docs/bots-guide.md

```diff
@@ -0,0 +1,192 @@
+        messages with "@followup".
+        Before running this, make sure to create a stream
+        called "followup" that your API user can send to.
+        '''
+```
+
+### triage_message(self, message)
+Is called by the system when a message was sent.
+#### Arguments
+* self - the instance the method is called on.
+* message - an object containing information about the message, e.g.
+ * content - the content of the message
+ * content_type - the type of the content, e.g. *'text/x-markdown'* for normal messages
+ * display_recipient - the name of the stream the message is sent to (string)
+ * is_mentioned - is the bot pinged with an '@' in the message? (boolean)
+ * sender_email - email of the sender (string)
+ * sender_full_name - full name of the sender (string)
+ * subject - topic of the message (string)
+ * timestamp - when was the message sent (integer)
+
+ ####Return values
+ * True if the bot should react to this message
+ * False otherwise
+
+ ####Example implementation
+```
+def triage_message(self, message):
+original_content = message['content']
+    if message['display_recipient'] == 'followup':
+        return False
+    is_follow_up = (original_content.startswith('@followup') or
+                        original_content.startswith('@follow-up'))
+    return is_follow_up
+```
+
+### handle_message(self, message, client)
+Is called when `triage_message` returns true, handles user message.
+#### Arguments
+* self - the instance the method is called on.
+* message - an object containing information about the message, e.g.
+ * content - the content of the message
+ * content_type - the type of the content, e.g. *'text/x-markdown'* for normal messages
+ * display_recipient - the name of the stream the message is sent to (string)
+ * is_mentioned - is the bot pinged with an '@' in the message? (boolean)
+ * sender_email - email of the sender (string)
+ * sender_full_name - full name of the sender (string)
+ * subject - topic of the message (string)
+ * timestamp - when was the message sent (integer)
+* client - used to interact with the server, e.g. to send a message
+ * use client.send_message(message) to send a message
+* state_handler - used to save states/information of the bot**beta**
+ * use `state_handler.set_state(state)` to set a state (any object)
+ * use `state_handler.get_state()` to retrieve the state set. Returns a `NoneType` object if no state is set.
+
+ ####Return values
+ None.
+
+ ####Example implementation
+ ```
+ def handle_message(self, message, client, state_handler):
+    original_content = message['content']
+    original_sender = message['sender_email']
+    new_content = original_content.replace('@followup',
+                                           'from %s:' % (original_sender,))
+
+    client.send_message(dict(
+        type='stream',
+        to='followup',
+        subject=message['sender_email'],
+        content=new_content,
+    ))
+ ```
+
+ ###Configuration file
```

Comment:
This indentation is also off, maybe because there is a missing whitespace after `###`?